src/RemoteInboxNotifications/EvaluationLogger.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace Automattic\WooCommerce\Admin\RemoteInboxNotifications;
+
+/**
+ * Class EvaluationLogger
+ *
+ * @package Automattic\WooCommerce\Admin\RemoteInboxNotifications
+ */
+class EvaluationLogger {
+	/**
+	 * Slug of the spec.
+	 *
+	 * @var string
+	 */
+	private $slug;
+
+	/**
+	 * Results of rules in the given spec.
+	 *
+	 * @var array
+	 */
+	private $results = array();
+
+	/**
+	 * Logger class to use.
+	 *
+	 * @var WC_Logger_Interface|null
+	 */
+	private $logger;
+
+	/**
+	 * Logger source.
+	 *
+	 * @var string logger source.
+	 */
+	private $source = 'remote-inbox-notifications';
```

Comment:
I think should not have a default value since it will be used for all rules.

Reply:
Since the logger works without the default source, I've removed it https://github.com/woocommerce/woocommerce-admin/pull/7194/commits/d4015c96bb31c48fd2153e50c9dca88ff7957114

The default log filename (log-YYYY-MM-DD-:random.log) will be used if `source` is omitted. 

The source can be overridden by passing `source` to `$logger_args`.

````
$evaluated_result = $rule_evaluator->evaluate( 
	$spec->rules, 
	$stored_state, 
	array( 
		'slug' => $spec->slug, 
		'source' => 'test'  
	) 
);
```